mevislab.github.io/content/tutorials/image_processing/image_processing6.md

```diff
@@ -4,194 +4,174 @@ date: 2025-05-05
 
+You have to select the right index for the RTSTRUCT. In our example it is index 2.
+Now add `DicomImportExtraOutput`module as showen in the below figure :
 
-👉As we can see here, there are no contours displayed in the image yet, so we need to add them:
+![Selecting CT and RTSTRUCT in DicomImport](/images/tutorials/image_processing/SelectingC.png "Selecting CT and RTSTRUCT in DicomImport")
 
 
-From `DicomImportExtraOutput` change using the arrow to 2/4 to show the contour as it show in the figure:
+### **Visualize RTSTRUCTs as colored CSOs:**
 
-![panel DicomImportExtraOutput to show the contourD](/images/tutorials/image_processing/panelDicomImportExtraOutputtoshowthecontour.png "panel DicomImportExtraOutput to show the contour")
+Now we need  `ExtractRTStruct`module to converts RTSTRUCT data into CSOs (Contour Segmentation Objects). CSOs allow MeVisLab to display and interact with the contours on the CT scan.
 
+Then coonect it with the `DicomImportExtraOutput` as shown in the figure :
 
-Now open the View2D panel, its will show the contour with numbers as it’s shown in the below figure :
+![ExtractRTStruct vonverter](/images/tutorials/image_processing/converter.png "ExtractRTStruct converter")
 
-![panel of View2D with contouring and numbers](/images/tutorials/image_processing/panelofView2Dwithcontouringandnumbers.png "panel of View2D with contouring and numbers")
 
+Add `SoView2DCSOExtensibleEdito ` to enable visualization and interaction with the CSOs in the 2D viewer, allowing you to edit and work with the contours directly. Connect it with `View2D`D and `ExtractRTStruct`, as shown, where `View2D` displays the CT scan with the contours.
 
-**Show labels next to each RTSTRUCT contour:**
+![Connecting SoView2DCSOExtensibleEditor](/images/tutorials/image_processing/cotour.png "Connecting SoView2DCSOExtensibleEditor")
 
-Open `CSOLabelRenderer panel`, its will appear in the main **(Label Code)**, we will add to it this line   
-{ `labelString = cso.getGroupAt(0).getLabel`() }, then press apply as shown in the figure:
+There are no names for the contours by default, so we need the `CSOLabelRenderer` module to add labels (e.g., 'Bladder', 'Prostate') next to each contour, helping to clearly identify the anatomy. The figure below shows that :
 
-![panel CSOLabelRenderer to name the contour](/images/tutorials/image_processing/panelCSOLabelRenderertonamethecontour..png "panel CSOLabelRenderer to name the contour")
+![CSOLabelRenderer](/images/tutorials/image_processing/CSOLabelRenderer.png " CSOLabelRenderer")
 
+As you can see, the contours are labeled with numbers, so we need to display proper names. To do this, open the `CSOLabelRenderer` panel — it will display the panel shown below.
```

Comment:
...we want to show the names for the contour to identify the segmented structure